src/trex/dataset.py

```diff
@@ -62,42 +73,56 @@ def read_all(
         assert n_transcriptome == len(names)
 
         if n_transcriptome == 1:
-            reads = self.read_one(
-                transcriptome_inputs[0],
-                self.output_dir / "entries.bam",
-                cell_id_tag,
-                require_umis,
-            )
-            if n_amplicon == 1:
+            files = self.find_paths(transcriptome_inputs[0], cell_id_tag)
+            reads = []
+            for file in files:
                 reads.extend(
                     self.read_one(
-                        amplicon_inputs[0],
-                        self.output_dir / "amplicon_entries.bam",
+                        file,
+                        self.output_dir / "entries.bam",
                         cell_id_tag,
                         require_umis,
                     )
                 )
+            if n_amplicon == 1:
+                files = self.find_paths(amplicon_inputs[0], cell_id_tag)
+                for file in files:
+                    reads.extend(
+                        self.read_one(
+                            amplicon_inputs[0],
+                            self.output_dir / "amplicon_entries.bam",
```

Comment:
It appears that the `amplicon_entries.bam` file gets overwritten every time `self.read_one` is called. It’ll only contain the reads from the last BAM file that was read in.

Reply:
the same should be true for `entries.bam` as well, right? (see line 73)

Reply:
I am now looking into this issue @marcelm. The path to creating a BAM file is: `read_all` -> `read_multiple` -> `read_bam` -> `read_alignment_file` -> output.bam
In the `read_multiple` function we have a loop that loops through all BAM files in a given path. As soon as the path contains more than one BAM file, calling the` read_alignment_file` function in the following loop will delete the previously created output file. 
I am looking for solutions now. One would be to append an existing BAM file but this is possible directly and not straightforward (one output BAM file per input file needs to be created, saved, all opened, and merged). 
This is an update on where I am currently. If you have a suggestion, Marcel, let me know.

Reply:
Found a solution: Instead of writing the new `entries.bam` file in the same function that also fetches and filters the reads, a separated `write_outbam()` function was created that writes the `entries.bam` output BAM file after the algorithm fetched all reads from all BAM files in the input path.